service_configuration_lib/spark_config.py

```diff
@@ -334,7 +339,20 @@ def get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
         log.warning(
             f'\nSetting spark.dynamicAllocation.maxExecutors as {max_executors}. If you wish to '
             f'change the value of maximum executors, please provide the exact value of '
-            f'spark.dynamicAllocation.maxExecutors in --spark-args\n',
+            f'spark.dynamicAllocation.maxExecutors in your spark args\n',
+        )
+
+    if 'jupyterhub' in spark_app_name and 'spark.dynamicAllocation.initialExecutors' not in spark_opts:
+        # set initialExecutors default equal to spark.executor.instances for Jupyter
+        initial_executors = int(spark_opts.get('spark.executor.instances', DEFAULT_EXECUTOR_INSTANCES))
+        # initialExecutors shouldn't be more than maxExecutors
+        initial_executors = min(initial_executors, int(spark_opts['spark.dynamicAllocation.maxExecutors']))
```

Comment:
Didn't we set initial executors as maxExecutors / 4?

Reply:
Yes but there's a condition on line 345. This logic is only for Jupyter Spark sessions